### PR TITLE
cleaner: add types

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Review by carlocab — APPROVED** [review]

File: Library/Homebrew/cleaner.rb [hunk 1]
```diff
@@ -48,7 +50,7 @@ def clean
     # [1]: https://github.com/Homebrew/brew/pull/11597
     # [2]: https://github.com/Homebrew/homebrew-core/issues/100190
     # [3]: https://github.com/Homebrew/brew/pull/13215
-    Dir.glob(@formula.info/"**/dir").each do |file|
+    Dir.glob("#{@formula.info}/**/dir").each do |file|
```
Comment (reitermarkus) [inline Library/Homebrew/cleaner.rb:53]: Why is this change necessary?

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: Library/Homebrew/cleaner.rb (hunk 1 above)
Comment (reitermarkus) [reply Library/Homebrew/cleaner.rb:53]: Add

```ruby
sig { returns(Formula) }
attr_reader :formula
```

instead of accessing the `@formula` variable directly, in case that's the reason.

Comment (cho-m) [reply Library/Homebrew/cleaner.rb:53]: It's because `Dir.glob` needs strings - https://ruby-doc.org/core-3.1.0/Dir.html#method-c-glob

An alternative would be `Pathname.glob` perhaps removing next line and doing,
```suggestion
    @formula.info.glob("**/dir").each do |info_dir_file|
```
Or
```suggestion
    Pathname.glob(@formula.info/"**/dir").each do |info_dir_file|
```

Comment (reitermarkus) [reply Library/Homebrew/cleaner.rb:53]: You can also do

```ruby
@formula.info.glob("**/dir")
```

Comment (reitermarkus) [reply Library/Homebrew/cleaner.rb:53]: You can also do

```suggestion
    @formula.info.glob("**/dir").each do |file|
```

Comment (cho-m) [reply Library/Homebrew/cleaner.rb:53]: That was what I was considering but also removing next line. `Pathname#glob` yields `Pathname`s so there is no need to do
```rb
      info_dir_file = Pathname(file)
```

**Review by reitermarkus — APPROVED** [review]

**Comment by reitermarkus** [comment]
Thanks!
